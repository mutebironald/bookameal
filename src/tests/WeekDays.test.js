import React from 'react'
import WeekDays from '../components/Weekdays'

describe("WeekDays component", () => {
    let wrapper;
    let mockFn;
    beforeEach(() => {
        mockFn = jest.fn();
        wrapper = mount(<WeekDays getMenu={mockFn}/>);
    });
    it("renders successfully", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should show meals on clicking", () => {
        wrapper
            .find("#monday")
            .first()
            .simulate("click");

        expect(mockFn.mock.calls.length).toBe(1);
    });
});
