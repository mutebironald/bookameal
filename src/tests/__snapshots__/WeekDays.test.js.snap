// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekDays component renders successfully 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Days
    getMenu={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "getMenu": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": <li
          class="list-group-item"
>
          <button
                    id="monday"
                    class="btn btn-block btn-lg"
                    data-day="monday"
          >
                    Monday
          </button>
</li>,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-block btn-lg"
            data-day="monday"
            id="monday"
            onClick={[Function]}
>
            Monday
</button>,
          "className": "list-group-item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <button
              id="monday"
              class="btn btn-block btn-lg"
              data-day="monday"
>
              Monday
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Monday",
              "className": "btn btn-block btn-lg",
              "data-day": "monday",
              "id": "monday",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Monday",
            ],
            "type": "button",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": <li
          class="list-group-item"
>
          <button
                    id="tuesday"
                    class="btn btn-block btn-lg"
                    data-day="tuesday"
          >
                    Tuesday
          </button>
</li>,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-block btn-lg"
            data-day="tuesday"
            id="tuesday"
            onClick={[Function]}
>
            Tuesday
</button>,
          "className": "list-group-item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <button
              id="tuesday"
              class="btn btn-block btn-lg"
              data-day="tuesday"
>
              Tuesday
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tuesday",
              "className": "btn btn-block btn-lg",
              "data-day": "tuesday",
              "id": "tuesday",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Tuesday",
            ],
            "type": "button",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": <li
          class="list-group-item"
>
          <button
                    id="wednesday"
                    class="btn btn-block btn-lg"
                    data-day="wednesday"
          >
                    Wednesday
          </button>
</li>,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-block btn-lg"
            data-day="wednesday"
            id="wednesday"
            onClick={[Function]}
>
            Wednesday
</button>,
          "className": "list-group-item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <button
              id="wednesday"
              class="btn btn-block btn-lg"
              data-day="wednesday"
>
              Wednesday
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Wednesday",
              "className": "btn btn-block btn-lg",
              "data-day": "wednesday",
              "id": "wednesday",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Wednesday",
            ],
            "type": "button",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": <li
          class="list-group-item"
>
          <button
                    id="thursday"
                    class="btn btn-block btn-lg"
                    data-day="thursday"
          >
                    Thursday
          </button>
</li>,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-block btn-lg"
            data-day="thursday"
            id="thursday"
            onClick={[Function]}
>
            Thursday
</button>,
          "className": "list-group-item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <button
              id="thursday"
              class="btn btn-block btn-lg"
              data-day="thursday"
>
              Thursday
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Thursday",
              "className": "btn btn-block btn-lg",
              "data-day": "thursday",
              "id": "thursday",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Thursday",
            ],
            "type": "button",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": <li
          class="list-group-item"
>
          <button
                    id="friday"
                    class="btn btn-block btn-lg"
                    data-day="friday"
          >
                    Friday
          </button>
</li>,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-block btn-lg"
            data-day="friday"
            id="friday"
            onClick={[Function]}
>
            Friday
</button>,
          "className": "list-group-item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <button
              id="friday"
              class="btn btn-block btn-lg"
              data-day="friday"
>
              Friday
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Friday",
              "className": "btn btn-block btn-lg",
              "data-day": "friday",
              "id": "friday",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Friday",
            ],
            "type": "button",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": <li
          class="list-group-item"
>
          <button
                    id="saturday"
                    class="btn btn-block btn-lg"
                    data-day="saturday"
          >
                    Saturday
          </button>
</li>,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-block btn-lg"
            data-day="saturday"
            id="saturday"
            onClick={[Function]}
>
            Saturday
</button>,
          "className": "list-group-item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <button
              id="saturday"
              class="btn btn-block btn-lg"
              data-day="saturday"
>
              Saturday
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Saturday",
              "className": "btn btn-block btn-lg",
              "data-day": "saturday",
              "id": "saturday",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Saturday",
            ],
            "type": "button",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": <li
          class="list-group-item"
>
          <button
                    id="sunday"
                    class="btn btn-block btn-lg"
                    data-day="sunday"
          >
                    Sunday
          </button>
</li>,
        "key": "6",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-block btn-lg"
            data-day="sunday"
            id="sunday"
            onClick={[Function]}
>
            Sunday
</button>,
          "className": "list-group-item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <button
              id="sunday"
              class="btn btn-block btn-lg"
              data-day="sunday"
>
              Sunday
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sunday",
              "className": "btn btn-block btn-lg",
              "data-day": "sunday",
              "id": "sunday",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Sunday",
            ],
            "type": "button",
          },
        ],
        "type": "li",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "getMenu": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <li
            class="list-group-item"
>
            <button
                        id="monday"
                        class="btn btn-block btn-lg"
                        data-day="monday"
            >
                        Monday
            </button>
</li>,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-block btn-lg"
              data-day="monday"
              id="monday"
              onClick={[Function]}
>
              Monday
</button>,
            "className": "list-group-item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <button
                id="monday"
                class="btn btn-block btn-lg"
                data-day="monday"
>
                Monday
</button>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Monday",
                "className": "btn btn-block btn-lg",
                "data-day": "monday",
                "id": "monday",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Monday",
              ],
              "type": "button",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": <li
            class="list-group-item"
>
            <button
                        id="tuesday"
                        class="btn btn-block btn-lg"
                        data-day="tuesday"
            >
                        Tuesday
            </button>
</li>,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-block btn-lg"
              data-day="tuesday"
              id="tuesday"
              onClick={[Function]}
>
              Tuesday
</button>,
            "className": "list-group-item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <button
                id="tuesday"
                class="btn btn-block btn-lg"
                data-day="tuesday"
>
                Tuesday
</button>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tuesday",
                "className": "btn btn-block btn-lg",
                "data-day": "tuesday",
                "id": "tuesday",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Tuesday",
              ],
              "type": "button",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": <li
            class="list-group-item"
>
            <button
                        id="wednesday"
                        class="btn btn-block btn-lg"
                        data-day="wednesday"
            >
                        Wednesday
            </button>
</li>,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-block btn-lg"
              data-day="wednesday"
              id="wednesday"
              onClick={[Function]}
>
              Wednesday
</button>,
            "className": "list-group-item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <button
                id="wednesday"
                class="btn btn-block btn-lg"
                data-day="wednesday"
>
                Wednesday
</button>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Wednesday",
                "className": "btn btn-block btn-lg",
                "data-day": "wednesday",
                "id": "wednesday",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Wednesday",
              ],
              "type": "button",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": <li
            class="list-group-item"
>
            <button
                        id="thursday"
                        class="btn btn-block btn-lg"
                        data-day="thursday"
            >
                        Thursday
            </button>
</li>,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-block btn-lg"
              data-day="thursday"
              id="thursday"
              onClick={[Function]}
>
              Thursday
</button>,
            "className": "list-group-item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <button
                id="thursday"
                class="btn btn-block btn-lg"
                data-day="thursday"
>
                Thursday
</button>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Thursday",
                "className": "btn btn-block btn-lg",
                "data-day": "thursday",
                "id": "thursday",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Thursday",
              ],
              "type": "button",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": <li
            class="list-group-item"
>
            <button
                        id="friday"
                        class="btn btn-block btn-lg"
                        data-day="friday"
            >
                        Friday
            </button>
</li>,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-block btn-lg"
              data-day="friday"
              id="friday"
              onClick={[Function]}
>
              Friday
</button>,
            "className": "list-group-item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <button
                id="friday"
                class="btn btn-block btn-lg"
                data-day="friday"
>
                Friday
</button>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Friday",
                "className": "btn btn-block btn-lg",
                "data-day": "friday",
                "id": "friday",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Friday",
              ],
              "type": "button",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": <li
            class="list-group-item"
>
            <button
                        id="saturday"
                        class="btn btn-block btn-lg"
                        data-day="saturday"
            >
                        Saturday
            </button>
</li>,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-block btn-lg"
              data-day="saturday"
              id="saturday"
              onClick={[Function]}
>
              Saturday
</button>,
            "className": "list-group-item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <button
                id="saturday"
                class="btn btn-block btn-lg"
                data-day="saturday"
>
                Saturday
</button>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Saturday",
                "className": "btn btn-block btn-lg",
                "data-day": "saturday",
                "id": "saturday",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Saturday",
              ],
              "type": "button",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": <li
            class="list-group-item"
>
            <button
                        id="sunday"
                        class="btn btn-block btn-lg"
                        data-day="sunday"
            >
                        Sunday
            </button>
</li>,
          "key": "6",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-block btn-lg"
              data-day="sunday"
              id="sunday"
              onClick={[Function]}
>
              Sunday
</button>,
            "className": "list-group-item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <button
                id="sunday"
                class="btn btn-block btn-lg"
                data-day="sunday"
>
                Sunday
</button>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sunday",
                "className": "btn btn-block btn-lg",
                "data-day": "sunday",
                "id": "sunday",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Sunday",
              ],
              "type": "button",
            },
          ],
          "type": "li",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
